aws/internal/keyvaluetags/service_generation_customizations.go

```diff
@@ -818,6 +831,17 @@ func ServiceTagType(serviceName string) string {
 	}
 }
 
+// ServiceTagType2 determines if the service tagging has a second tag type.
+// The two types must be equivalent.
+func ServiceTagType2(serviceName string) string {
+	switch serviceName {
+	case "ec2":
```

Comment:
👍 
Will allow us to get rid of `ec2TagsFromTagDescriptions`:

https://github.com/terraform-providers/terraform-provider-aws/blob/8f067bd6e4cf11bbce393dafb56b679901f5f024/aws/tags.go#L38-L54

Reply:
We could also get rid of `ec2TagSpecificationsFromMap` by adding

```go
func (tags KeyValueTags) Ec2TagSpecifications(resourceType string) []*ec2.TagSpecification {
```

but that's another issue.